bugbear.py

```diff
@@ -1104,6 +1105,29 @@ def check_for_b022(self, node):
         ):
             self.errors.append(B022(node.lineno, node.col_offset))
 
+    def check_for_b908(self, node: ast.With):
+        has_multiline_body = len(node.body) > 1
+        for node_item in node.items:
+            if (
+                not hasattr(node_item.context_expr, "func")
+                or not hasattr(node_item.context_expr.func, "value")
+                or not hasattr(node_item.context_expr.func.value, "id")
+            ):
+                continue
+            if (
+                # pytest form
+                node_item.context_expr.func.value.id == "pytest"
+                and node_item.context_expr.func.attr == "raises"
+            ) or (
+                # unittest form
+                node_item.context_expr.func.value.id == "self"
+                # to catch all variants like
+                # 'assertRaises', 'assertRaisesRegex', 'assertRaisesRegexp'
+                and node_item.context_expr.func.attr.startswith("assert")
```

Comment:
I'd prefer to list all the names explicitly, since it's plausible people will have other `with self.assert...` like methods that don't work like the unittest ones. I think the list you have is good, but I'd also add `assertWarns` and its two regex forms.